From 2e3c5bd212ddc1ec1d3caf81f71e1fa08df1433e Mon Sep 17 00:00:00 2001
From: Alvin Chang <alvinga@andestech.com>
Date: Fri, 21 Feb 2025 22:17:27 +0800
Subject: [RISE 06/10] core: riscv: Move ecall macros to public sbi.h header
 file

Other modules may invoke ecall to trigger SBI request. Thus, this commit
moves ecall macros from private C source file to public sbi.h header
file.

Signed-off-by: Alvin Chang <alvinga@andestech.com>
---
 core/arch/riscv/include/sbi.h | 23 +++++++++++++++++++++++
 core/arch/riscv/kernel/sbi.c  | 23 -----------------------
 2 files changed, 23 insertions(+), 23 deletions(-)

diff --git a/core/arch/riscv/include/sbi.h b/core/arch/riscv/include/sbi.h
index 753e7372dffc..f6e503514044 100644
--- a/core/arch/riscv/include/sbi.h
+++ b/core/arch/riscv/include/sbi.h
@@ -28,6 +28,29 @@
 
 #ifndef __ASSEMBLER__
 
+struct sbiret {
+	long error;
+	long value;
+};
+
+#define _sbi_ecall(ext, fid, arg0, arg1, arg2, arg3, arg4, arg5, ...) ({  \
+	register unsigned long a0 asm("a0") = (unsigned long)arg0; \
+	register unsigned long a1 asm("a1") = (unsigned long)arg1; \
+	register unsigned long a2 asm("a2") = (unsigned long)arg2; \
+	register unsigned long a3 asm("a3") = (unsigned long)arg3; \
+	register unsigned long a4 asm("a4") = (unsigned long)arg4; \
+	register unsigned long a5 asm("a5") = (unsigned long)arg5; \
+	register unsigned long a6 asm("a6") = (unsigned long)fid;  \
+	register unsigned long a7 asm("a7") = (unsigned long)ext;  \
+	asm volatile ("ecall" \
+		    : "+r" (a0), "+r" (a1) \
+		    : "r" (a2), "r" (a3), "r" (a4), "r" (a5), "r"(a6), "r"(a7) \
+		    : "memory"); \
+	(struct sbiret){ .error = a0, .value = a1 }; \
+})
+
+#define sbi_ecall(...)	_sbi_ecall(__VA_ARGS__, 0, 0, 0, 0, 0, 0, 0)
+
 /* SBI function IDs for Base extension */
 enum sbi_ext_base_fid {
 	SBI_EXT_BASE_GET_SPEC_VERSION = 0,
diff --git a/core/arch/riscv/kernel/sbi.c b/core/arch/riscv/kernel/sbi.c
index 06d297e9be1e..6ba191b6dc6e 100644
--- a/core/arch/riscv/kernel/sbi.c
+++ b/core/arch/riscv/kernel/sbi.c
@@ -6,29 +6,6 @@
 #include <riscv.h>
 #include <sbi.h>
 
-struct sbiret {
-	long error;
-	long value;
-};
-
-#define _sbi_ecall(ext, fid, arg0, arg1, arg2, arg3, arg4, arg5, ...) ({  \
-	register unsigned long a0 asm("a0") = (unsigned long)arg0; \
-	register unsigned long a1 asm("a1") = (unsigned long)arg1; \
-	register unsigned long a2 asm("a2") = (unsigned long)arg2; \
-	register unsigned long a3 asm("a3") = (unsigned long)arg3; \
-	register unsigned long a4 asm("a4") = (unsigned long)arg4; \
-	register unsigned long a5 asm("a5") = (unsigned long)arg5; \
-	register unsigned long a6 asm("a6") = (unsigned long)fid;  \
-	register unsigned long a7 asm("a7") = (unsigned long)ext;  \
-	asm volatile ("ecall" \
-		: "+r" (a0), "+r" (a1) \
-		: "r" (a2), "r" (a3), "r" (a4), "r" (a5), "r"(a6), "r"(a7) \
-		: "memory"); \
-	(struct sbiret){ .error = a0, .value = a1 }; \
-})
-
-#define sbi_ecall(...) _sbi_ecall(__VA_ARGS__, 0, 0, 0, 0, 0, 0, 0)
-
 /**
  * sbi_probe_extension() - Check if an SBI extension ID is supported or not.
  * @extid: The extension ID to be probed.
-- 
2.43.0

